Sources/ScipioKit/Executor.swift

```diff
@@ -84,59 +82,136 @@ Unknown error occurered.
+        let errorHandle = stderrPipe.fileHandleForReading
+
+        let outputBuffer = ProcessOutputBuffer()
+        let errorBuffer = ProcessOutputBuffer()
+
+        let localStreamOutput = streamOutput
+        let localDecoder = decoder
+
+        let outputTask = Task {
+            for try await data in outputHandle.byteStream() {
+                let bytes = [UInt8](data)
+                await localStreamOutput?(bytes)
+                await outputBuffer.append(bytes)
             }
-        )
+        }
 
-        let process = Process(
-            arguments: arguments,
-            outputRedirection: outputRedirection
-        )
+        let errorOutputTask = Task {
+            for try await data in errorHandle.byteStream() {
+                let bytes = [UInt8](data)
+                await errorBuffer.append(bytes)
+            }
+        }
 
-        var result: ProcessResult
         do {
-            try process.launch()
-            result = try await process.waitUntilExit()
+            try process.run()
         } catch {
             throw ProcessExecutorError.unknownError(error)
         }
 
-        // respects failure state
-        result.setOutput(result.output.map { _ in outputBuffer })
-        result.setStderrOutput(result.stderrOutput.map { _ in errorBuffer })
+        try await outputTask.value
+        try await errorOutputTask.value
 
-        switch result.exitStatus {
-        case .terminated(let code) where code == 0:
-            return result
-        case .terminated:
-            let errorOutput = try? decoder.decode(result)
-            throw ProcessExecutorError.terminated(errorOutput: errorOutput)
-        case .signalled(let signal):
-            throw ProcessExecutorError.signalled(signal)
+        // Wait for process to complete
+        return try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<ExecutorResult, Error>) in
+            process.terminationHandler = { process in
+                Task {
+                    // Clean up handlers
+                    outputHandle.readabilityHandler = nil
+                    errorHandle.readabilityHandler = nil
+
+                    // Read any remaining data
+                    let remainingStdout = outputHandle.readDataToEndOfFile()
+                    let remainingStderr = errorHandle.readDataToEndOfFile()
+
+                    if !remainingStdout.isEmpty {
+                        let bytes = [UInt8](remainingStdout)
+                        await localStreamOutput?(bytes)
+                        await outputBuffer.append(bytes)
+                    }
+
+                    if !remainingStderr.isEmpty {
+                        let bytes = [UInt8](remainingStderr)
+                        await errorBuffer.append(bytes)
+                    }
+
+                    // Close file handles
+                    try? outputHandle.close()
+                    try? errorHandle.close()
```

Comment:
```suggestion
```

We don't need to close `fileHandleForReading` manually.

- https://developer.apple.com/documentation/foundation/pipe/filehandleforreading
  - > The descriptor represented by this object is deleted, and the object itself is automatically deallocated when the receiver is deallocated.
  - > You don’t need to send [closeFile()](https://developer.apple.com/documentation/foundation/filehandle/closefile()) to this object or explicitly release the object after you have finished using it.